server/routes/profile.js

```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const router = express.Router();
+const { allProfiles, 
+        getProfileById, 
+        addProfile, 
+        updateProfile }
+ = require("../controllers/profile");
+ const { validateProfile, validateProfileId } = require("../validate");
+ const protect =  require("../middleware/auth");
+
+router.route("/").get(allProfiles);
+
+router.route("/:id").get(validateProfileId, getProfileById);
+
+router.route("/").post(validateProfile, addProfile);
```

Comment:
```suggestion
router.route("/").post(protect, validateProfile, addProfile);
```

Only authorized users should be able to create a profile